optuna/study/_optimize.py

```diff
@@ -122,60 +229,6 @@ def _optimize(
         progress_bar.close()
 
 
-def _optimize_sequential(
-    study: "optuna.Study",
-    func: "optuna.study.study.ObjectiveFuncType",
-    n_trials: Optional[int],
-    timeout: Optional[float],
-    catch: Tuple[Type[Exception], ...],
-    callbacks: Optional[List[Callable[["optuna.Study", FrozenTrial], None]]],
-    gc_after_trial: bool,
-    reseed_sampler_rng: bool,
-    time_start: Optional[datetime.datetime],
-    progress_bar: Optional[pbar_module._ProgressBar],
-) -> None:
-    if reseed_sampler_rng:
-        study.sampler.reseed_rng()
-
-    i_trial = 0
-
-    if time_start is None:
-        time_start = datetime.datetime.now()
-
-    while True:
-        if study._stop_flag:
-            break
-
-        if n_trials is not None:
-            if i_trial >= n_trials:
-                break
-            i_trial += 1
-
-        if timeout is not None:
-            elapsed_seconds = (datetime.datetime.now() - time_start).total_seconds()
-            if elapsed_seconds >= timeout:
-                break
-
-        try:
-            frozen_trial = _run_trial(study, func, catch)
-        finally:
-            # The following line mitigates memory problems that can be occurred in some
-            # environments (e.g., services that use computing containers such as CircleCI).
-            # Please refer to the following PR for further details:
-            # https://github.com/optuna/optuna/pull/325.
-            if gc_after_trial:
-                gc.collect()
-
-        if callbacks is not None:
-            for callback in callbacks:
-                callback(study, frozen_trial)
-
-        if progress_bar is not None:
-            progress_bar.update((datetime.datetime.now() - time_start).total_seconds())
-
-    study._storage.remove_session()
-
-
```

Comment:
Could you revert this change? It makes reviews harder.

Reply:
I changed the order so that I could use `_optimize_sequential` as a default parameter for `_optimize` function